atmcorr/processing.py

```diff
@@ -43,6 +44,62 @@ def _earthdata_credentials_from_env():
     return auth
 
 
+def _get_tile_index_iter_interpolation(tiling_shape):
+    # make sure to pick corners and center values
+    indices = [(0, 0),
+               (0, tiling_shape[1] - 1),
+               (tiling_shape[0] - 1, 0),
+               (tiling_shape[0] - 1, tiling_shape[1] - 1),
+               (int(tiling_shape[0] / 2), int(tiling_shape[1] / 2))
+               ]
+    # pick every fifth element column/row wise
+    all_indices = list(itertools.product(range(0, tiling_shape[0], 5),
+                                         range(0, tiling_shape[1], 5))) + indices
+    for i in all_indices:
+        yield i
+
+
+def _get_tile_index_iter(tiling_shape):
+    return itertools.product(range(tiling_shape[0]), range(tiling_shape[1]))
+
+
+def _interpolate_correction_params(correction_params, geometry_dict):
+    # interpolate xa and xc
+    for field in ['xa', 'xc']:
+        for band in range(correction_params[field].shape[0]):
+            selected = correction_params[field][band, :, :].copy()
+            x_idx, y_idx = np.mgrid[:selected.shape[0], :selected.shape[1]]
+            xyi = np.column_stack([x_idx.ravel(), y_idx.ravel()])
+            subset_mask = np.where(~np.isnan(selected))
+            known_data = []
+            xy = np.zeros((len(subset_mask[0]), 2))
+            for idx,  index in enumerate(zip(*subset_mask)):
+                xy[idx] = index
+                known_data.append(selected[index])
+            interpolated = scipy.interpolate.griddata(xy, known_data, xyi, method='nearest')
+            interpolated = interpolated.reshape((selected.shape))
+            correction_params[field][band, :, :] = interpolated.copy()
+
+    # interploate xb correction params in the segments defined by the sensor azimuth
+    seg_a = geometry_dict['sensor_azimuth'] <= np.mean(geometry_dict['sensor_azimuth'])
+    seg_b = geometry_dict['sensor_azimuth'] > np.mean(geometry_dict['sensor_azimuth'])
+    segments = [seg_a, seg_b]
+    for segment in segments:
+        for band in range(correction_params['xb'].shape[0]):
+            selected = correction_params['xb'][band, :, :].copy()
+            segment_mask = np.where(segment)
+            xyi = np.asarray(list(map(list, zip(*segment_mask))))
+            subset_mask = np.where(~np.isnan(selected) & segment)
```

Comment:
I think we could speed this up with boolean indices and get rid of the for loop and appending data to the `known_data` list. Is it worth (performance-wise) to look into that?